InnerEye/ML/configs/classification/CovidModel.py

```diff
@@ -237,45 +239,118 @@ def compute_and_log_metrics(self,
     def generate_custom_report(self, report_dir: Path, model_proc: ModelProcessing) -> Path:
         """
         Generate a custom report for the Covid model. This report will read the file model_output.csv generated for
-        the training, validation or test sets and compute the multiclass accuracy based on this.
+        the training, validation or test sets and compute both the multiclass accuracy and the accuracy for each of the
+        hierarchical tasks.
         :param report_dir: Directory report is to be written to
         :param model_proc: Whether this is a single or ensemble model (model_output.csv will be located in different
         paths for single vs ensemble runs.)
         """
 
+        label_prefix = LoggingColumns.Label.value
+        output_prefix = LoggingColumns.ModelOutput.value
+
         def get_output_csv_path(mode: ModelExecutionMode) -> Path:
             p = get_best_epoch_results_path(mode=mode, model_proc=model_proc)
             return self.outputs_folder / p / MODEL_OUTPUT_CSV
 
         def get_labels_and_predictions(df: pd.DataFrame) -> pd.DataFrame:
+            """
+            Given a dataframe with predictions for a single subject, returns the label and model output for the
+            tasks: CVX03vs12, CVX0vs3, CVX1vs2 and multiclass.
+            """
             labels = []
             predictions = []
             for target in self.target_names:
                 target_df = df[df[LoggingColumns.Hue.value] == target]
-                predictions.append(target_df[LoggingColumns.ModelOutput.value])
-                labels.append(target_df[LoggingColumns.Label.value])
+                predictions.append(target_df[output_prefix].item())
+                labels.append(target_df[label_prefix].item())
 
-            return pd.DataFrame.from_dict({LoggingColumns.Patient.value: [df.iloc[0][LoggingColumns.Patient.value]],
-                                           LoggingColumns.ModelOutput.value: [np.argmax(predictions)],
-                                           LoggingColumns.Label.value: [np.argmax(labels)]})
+            pred_cvx03vs12 = predictions[1] + predictions[2]
+            label_cvx03vs12 = 1 if labels[1] or labels[2] else 0
```

Comment:
minor: for this kind of statement I believe the intent would be clearer with `int(labels[1] or labels[2])`

Reply:
sorry, I meant `label_cvx03vs12 = int(labels[1] or labels[2])`. on second thought it may be better to revert to how it was initially, for consistency with the rest 🙂